### DISCUSSION TITLE
How to implement linear heatmap

### DISCUSSION THREAD

**Discussion by a0ax** [maplibre/maplibre-gl-js#3884] [Q&A]
I'm looking to build something that takes GeoJSON input and creates a linear kind of heatmap, like the example below, using MapLibre.
![310917551-6c657f93-d10e-4acf-b6d1-2543bb167bdd](https://github.com/maplibre/maplibre-gl-js/assets/61996958/d90b0a3d-07d4-4821-aeb5-4704e7c34169)
I.e. the density of traces at a point defines the color at that point. But I truly have no idea how to do this--where do I begin? How can I individually style "segments" of the line? Any and all information/help at all would really mean a lot.
Thanks in advance!

**Comment by wipfli** [comment]
Maybe with deckgl https://deck.gl/examples/line-layer ?

**Comment by IvanSanchez** [comment]
Been working on something like that for a while - see https://mastodon.social/@IvanSanchez/109824286426044519 and https://gitlab.com/IvanSanchez/gleo/-/blob/master/browser-demos/74-ais-drag-drop-heatstroke.html and https://ivan.sanchezortega.es/2023-gleo-slides/slide-42.html . No connector to maplibre I'm afraid.

**Comment by HarelM** [comment]
Please don't start a new discussion when you already have a different one open.
See here:
https://github.com/maplibre/maplibre-gl-js/discussions/3813
